docs/user-guide/03-convenience-features.md

```diff
@@ -149,7 +149,7 @@ int rs = foo.staticMethod(a,b);
 
 ### Mutability of member function
 
-For GPU performance considerations, the `this` argument in a member function is immutable by default. If you modify the content in `this` argument, the modification will be discarded after the call and does not affect the input object. If you intend to define a member function that mutates the object, use `[mutating]` attribute on the member function as shown in the following example.
+For GPU performance considerations, the `this` argument in a member function is immutable by default. Attempting to modify `this` will throw compile errors. If you intend to define a member function that mutates the object, use `[mutating]` attribute on the member function as shown in the following example.
```

Comment:
will throw -> will result in

throw is a word with special meaning and usually assoicated error codes or exceptions.